### PR TITLE
Fix SVE blendv mask handling after bool fill

### PR DESCRIPTION
**Pull request by robert-hardwick**
Stack from [ghstack](https://github.com/ezyang/ghstack/tree/0.15.0) (oldest at bottom):
* __->__ #183697

This fixes the remaining SVE SDPA rewriter failures after the VecMask bit-cast fix.

The failing path comes from compiled CPU vector code for boolean mask operations such as:

`mask.masked_fill(torch.logical_not(mask), -float("inf")).to(torch.float32)`

On SVE, blendv was reconstructing its predicate by checking whether each stored mask lane was exactly 0xffffffff. That is too strict: due to a GCC codegen bug being tracked separately, a logically true lane can be produced with a different nonzero bit pattern rather than the canonical all-bits-true value. When that happened, blendv misread the lane as false and selected the wrong value.

This changes SVE blendv predicate reconstruction from “equal to all-bits-true” to “not equal to zero”, matching the boolean-mask contract: zero is false, any nonzero value is true.

The PR also adds a targeted regression test for the boolean masked_fill(..., -inf).to(float32) pattern and removes the SVE256 xfails from the affected SDPA rewriter tests.

Fixes #177244

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @jerryzh168 @aditew01 @voznesenskym @penguinwu @EikanWang @Guobing-Chen @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben @jataylo

### CODE REVIEW

**Comment by aditew01** [comment]
@claude review the PR, check if the unit test coverage is enough for this and are the ci failures related

**Review by aditew01 — APPROVED** [review]
